pkg/apis/deployment/v1/architecture.go

```diff
@@ -87,3 +73,32 @@ func (a ArangoDeploymentArchitectureType) Validate() error {
 		return errors.Errorf("Unknown architecture type %s", q)
 	}
 }
+
+func (a ArangoDeploymentArchitectureType) AsNodeSelectorRequirement() core.NodeSelectorTerm {
+	return core.NodeSelectorTerm{
+		MatchExpressions: []core.NodeSelectorRequirement{
+			{
+				Key:      k8sutil.NodeArchAffinityLabel,
+				Operator: "In",
+				Values:   []string{string(a)},
+			},
+		},
+	}
+}
+
+func GetArchsFromNodeSelector(selectors []core.NodeSelectorTerm) ArangoDeploymentArchitecture {
```

Comment:
Non-api helper function in api? Move to k8sutils

Reply:
+ cover case of change from "beta.kubernetes.io/arch" to "kubernetes.io/arch" (it should be still silent)

Reply:
+ run unique on list

Reply:
+ sort them